pkg/quarterdeck/db/models/users.go

```diff
@@ -361,32 +363,56 @@ func (u *User) UpdateLastLogin(ctx context.Context) (err error) {
 }
 
 const (
-	userInviteSQL = "INSERT INTO user_invitations (organization_id, email, expires, token, secret, created_by, created, modified) VALUES (:orgID, :email, :expires, :token, :secret, :createdBy, :created, :modified)"
+	userInviteSQL = "INSERT INTO user_invitations (user_id, organization_id, role, email, expires, token, secret, created_by, created, modified) VALUES (:userID, :orgID, :role, :email, :expires, :token, :secret, :createdBy, :created, :modified)"
 )
 
 // Create an invitation in the database from the user to the specified email address
-// and return the invitation token to send to the user.
-func (u *User) Invite(ctx context.Context, email string) (token string, err error) {
+// and return the invitation token to send to the user. This method returns an error if
+// the invitee is already associated with the organization.
+func (u *User) CreateInvite(ctx context.Context, email, role string) (userInvite *UserInvitation, err error) {
 	var (
 		invite *db.VerificationToken
 	)
 
+	if role == "" {
+		return nil, ErrMissingInviteRole
+	}
+
 	// Create a token that expires in 7 days
 	if invite, err = db.NewVerificationToken(email); err != nil {
-		return "", err
+		return nil, err
+	}
+
+	// Get the invitee's ID if they already exist or create a new ID
+	var userID ulid.ULID
+	if user, err := GetUserEmail(ctx, email, ulid.ULID{}); err != nil {
+		if !errors.Is(err, ErrNotFound) {
+			return nil, err
+		}
+		userID = ulids.New()
```

Comment:
Some comments might be helpful here; so if we try to fetch their email and `err == nil` then you're expecting us to go to the `else if` block? It is a bit tricky to read.